src/main/java/org/aist/aide/mappingservice/domain/services/MappingCrudService.java

```diff
@@ -0,0 +1,80 @@
+package org.aist.aide.mappingservice.domain.services;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+import org.aist.aide.mappingservice.domain.exceptions.NotFoundException;
+import org.aist.aide.mappingservice.domain.exceptions.ValidationFailureException;
+import org.aist.aide.mappingservice.domain.models.Mapping;
+import org.aist.aide.mappingservice.service.repositories.MappingRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class MappingCrudService {
+    private static final Logger LOGGER = Logger.getLogger(MappingCrudService.class.getName());
+    private MappingRepository mappingRepository;
+
+    public MappingCrudService(@Autowired MappingRepository mappingRepository) {
+        this.mappingRepository = mappingRepository;
+    }
+
+    public List<Mapping> getMappings() {
+        List<Mapping> target = new ArrayList<>();
+        mappingRepository.findAll().forEach(target::add);
```

Comment:
finalAll doesn't return a list?

Reply:
For these type of repos it returns iterable... to avoid logic complications in the controllers I moved handling this over here.